x/margin/keeper/open_long.go

```diff
@@ -49,6 +60,11 @@ func (k Keeper) OpenLong(ctx sdk.Context, poolId uint64, msg *types.MsgOpen) (*t
 		return nil, err
 	}
 
+	// If the collateral asset is not usdc, custody amount equals to leverage amount
+	if msg.CollateralAsset != ptypes.USDC {
+		custodyAmount = leveragedAmount
+	}
```

Comment:
@kenta-elys and what happens to `custodyAmount` if collateral asset is equal to `USDC`?

Reply:
@cosmic-vagabond 
If collateral asset equals to USDC, it will just use the custodyAmount calculated in the above.